### PR TITLE
Video to pdf (CLI)

### LINKED ISSUE #1155: Video to PDF (CLI)
# Description

A script to convert a given video to a pdf file and save it.

## Language

- [x] Python

## Checklist:

- **Details on the Script**

| name            | about                                                  | labels  | assignees                                                      |
| --------------- | ------------------------------------------------------ | ------- | -------------------------------------------------------------- |
| video_to_pdf | A script to convert a video to pdf file | LGM-SOC | @tanvi355 |

**Describe the solution you'd like**: 

Extracting the audio from a given video, converting it into text and then save it as a pdf.

### PR DESCRIPTION
**Pull request by tanvi355**
# Description

A script to convert a video into a pdf file. It uses the MoviePy, FPDF and speech recognition modules.

Fixes #1155 

## Type of change

_Please delete options that are not relevant._

- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update

# Checklist:

- [x] My code follows the style guidelines(Clean Code) of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have created a helpful and easy to understand `README.md`
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests/screenshots(if any) that prove my fix is effective or that my feature works

### CODE REVIEW

File: Python/Video to PDF(CLI)/video_to_pdf.py [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+#variables
+video_clip = ''
+audio_clip = ''
+
+video_filepath = input('Enter video path : ')
+video_clip = VideoFileClip(str(video_filepath))
+
+
+#function to convert audio to pdf
+def audio_to_pdf():
```
Comment (RohiniRG) [inline Python/Video to PDF(CLI)/video_to_pdf.py:17]: Add docstrings to your functions explaining what its doing.

File: Python/Video to PDF(CLI)/video_to_pdf.py [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+        print("Conversion not performed")
+
+    #delete audio and text files
+    os.remove("my_text.txt")
+    os.remove("my_audio.wav")
+
+
+#function to convert text to pdf
+def text_to_pdf(file):
+    pdf = FPDF(format='letter', unit='in')
```
Comment (RohiniRG) [inline Python/Video to PDF(CLI)/video_to_pdf.py:46]: Add a docstring

File: Python/Video to PDF(CLI)/video_to_pdf.py [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+        pdf.multi_cell(effective_page_width, 0.15, x)
+        pdf.ln(1)
+
+    #save the pdf
+    pdf.output("../Video to PDF(CLI)/my_pdf.pdf")
+
+
+if __name__ == '__main__':
+    print('Starting conversion....')
+    audio_to_pdf()
```
Comment (RohiniRG) [inline Python/Video to PDF(CLI)/video_to_pdf.py:63]: Add a newline at the end of the file

File: Python/Video to PDF(CLI)/README.md [hunk 4]
```diff
@@ -0,0 +1,13 @@
+# Video to PDF
+Convert any video of your choice into a pdf file!
+
+## How to use this script?
+- Enter the file path of the video from your local machine.
+- The script will convert it into a pdf and save it as 'my_pdf.pdf' in the same location where the video was present.
```
Comment (RohiniRG) [inline Python/Video to PDF(CLI)/README.md:6]: Write the file name as ` my_pdf.pdf `

**Review by RohiniRG — CHANGES_REQUESTED** [review]
Nice work! I have requested a few changes

**Comment by tanvi355** [comment]
> Nice work! I have requested a few changes

I have made the changes and also added a try-except block because I missed the 'when no video is selected' case earlier.

File: Python/Video to PDF(CLI)/README.md [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+## How to use this script?
+- Enter the file path of the video from your local machine.
+- The script will convert it into a pdf and save it as `my_pdf.pdf` in the same location where the video was present.
+
+To install all the dependencies just run the command
+
+``
+pip install -r requirements.txt
+``
+- Sample Video : [Link](https://www.youtube.com/watch?v=rGHrKkieqCY)
```
Comment (HarshCasper) [inline Python/Video to PDF(CLI)/README.md:13]: Can you follow the Template README over the repo? 

Also, convert the sample video into a GIF, upload over Imgur and reference it here.

File: Python/Video to PDF(CLI)/README.md [hunk 6]
```diff
@@ -0,0 +1,13 @@
+# Video to PDF
+Convert any video of your choice into a pdf file!
+
+## How to use this script?
+- Enter the file path of the video from your local machine.
+- The script will convert it into a pdf and save it as `my_pdf.pdf` in the same location where the video was present.
```
Comment (HarshCasper) [inline Python/Video to PDF(CLI)/README.md:6]: Remove the PDF from the PR

File: Python/Video to PDF(CLI)/README.md [hunk 7]
```diff
@@ -0,0 +1,13 @@
...
+
+To install all the dependencies just run the command
+
+``
+pip install -r requirements.txt
+``
```
Comment (HarshCasper) [inline Python/Video to PDF(CLI)/README.md:12]: Add a usage instruction

**Review by HarshCasper — CHANGES_REQUESTED** [review]

**Review by RohiniRG — APPROVED** [review]
LGTM

File: Python/Video to PDF(CLI)/README.md [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+
+## Running the script
+- Run the script using `python video_to_pdf.py`.
+- Next, enter the file path of the video from your local machine.
+- The script will convert it into a pdf and save it as `my_pdf.pdf` in the same location where the video was present.
+
+## Input:
```
Comment (HarshCasper) [inline Python/Video to PDF(CLI)/README.md:26]: Give a line break

File: Python/Video to PDF(CLI)/README.md [hunk 9]
```diff
@@ -0,0 +1,33 @@
...
+![Gif](https://imgur.com/UPuFi4P.gif)
+
+## Output:
```
Comment (HarshCasper) [inline Python/Video to PDF(CLI)/README.md:29]: Give a line break

File: Python/Video to PDF(CLI)/README.md [hunk 10]
```diff
@@ -0,0 +1,33 @@
...
+![Image](https://imgur.com/sRmEiSg.png)
+
+## Author
```
Comment (HarshCasper) [inline Python/Video to PDF(CLI)/README.md:32]: Give a line break

File: Python/Video to PDF(CLI)/README.md (hunk 7 above)
Comment (HarshCasper) [reply Python/Video to PDF(CLI)/README.md:19]: Kindly resolve this by adding a usage instruction.

File: Python/Video to PDF(CLI)/README.md [hunk 11]
```diff
@@ -0,0 +1,33 @@
...
+- MoviePy
+- FPDF
+
+To install all these dependencies just run the command
+
+``
+pip install -r requirements.txt
+``
+
+## Running the script
```
Comment (HarshCasper) [inline Python/Video to PDF(CLI)/README.md:21]: Give a line break

File: Python/Video to PDF(CLI)/README.md [hunk 12]
```diff
@@ -0,0 +1,33 @@
+# Video to PDF
```
Comment (HarshCasper) [inline Python/Video to PDF(CLI)/README.md:1]: Give a line break

File: Python/Video to PDF(CLI)/README.md [hunk 13]
```diff
@@ -0,0 +1,33 @@
...
+The aim of this script is to convert a given video into a pdf.
+This script:
+1. Takes a video as input.
+2. Extracts audio from it.
+3. Converts the audio into text.
+4. Writes this text into a pdf and saves it.
```
Comment (HarshCasper) [inline Python/Video to PDF(CLI)/README.md:7]: Keep PDF is capitals and in bold

**Review by HarshCasper — CHANGES_REQUESTED** [review]

**Comment by iamakkkhil** [comment]
Any Updates?

**Comment by tanvi355** [comment]
> Any Updates?

Will do it by EOD.

File: Python/Video to PDF(CLI)/README.md (hunk 7 above)
Comment (iamakkkhil) [reply Python/Video to PDF(CLI)/README.md:19]: Add a ss of terminal where script is running.

File: Python/Video to PDF(CLI)/README.md [hunk 14]
```diff
@@ -0,0 +1,33 @@
...
+To install all these dependencies just run the command
+
+``
+pip install -r requirements.txt
+``
+
+## Running the script
+- Run the script using `python video_to_pdf.py`.
+- Next, enter the file path of the video from your local machine.
+- The script will convert it into a pdf and save it as `my_pdf.pdf` in the same location where the video was present.
```
Comment (iamakkkhil) [inline Python/Video to PDF(CLI)/README.md:24]: Add numbered points instead of bulleted points

**Review by iamakkkhil — CHANGES_REQUESTED** [review]
Looks good to me.

File: Python/Video to PDF(CLI)/README.md [hunk 15]
```diff
@@ -0,0 +1,41 @@
...
+The aim of this script is to convert a given video into a pdf.
+This script:
+
+1. Takes a video as input.
+2. Extracts audio from it.
+3. Converts the audio into text.
+4. Writes this text into a <strong>PDF</strong> and saves it.
+
+## Setup Instructions
+<br/>
```
Comment (vybhav72954) [inline Python/Video to PDF(CLI)/README.md:12]: Why are break tags here? By `line break` Harsh meant, an empty line

File: Python/Video to PDF(CLI)/video_to_pdf.py [hunk 16]
```diff
@@ -0,0 +1,78 @@
...
+        print("Conversion not performed")
+
+    #delete audio and text files
+    os.remove("my_text.txt")
+    os.remove("my_audio.wav")
+
+
+def text_to_pdf(file):
+    '''
+    Function to convert text to pdf:
```
Comment (vybhav72954) [inline Python/Video to PDF(CLI)/video_to_pdf.py:49]: This is not what a DocString actually is though.

Take this for [example](https://www.programiz.com/python-programming/docstrings). I am not going to nitpick here, so don't change it, for now, would be better if you keep this in mind in the future though.

File: Python/Video to PDF(CLI)/README.md [hunk 17]
```diff
@@ -0,0 +1,36 @@
+# Video to PDF
+
+The aim of this script is to convert a given video into a pdf.
```
Comment (HarshCasper) [inline Python/Video to PDF(CLI)/README.md:3]: Capitalize PDF.

File: Python/Video to PDF(CLI)/README.md [hunk 18]
```diff
@@ -0,0 +1,36 @@
...
+3. Converts the audio into text.
+4. Writes this text into a <strong>PDF</strong> and saves it.
+
+## Setup Instructions
+
+You will need the following 3 modules to be installed:
+- Speech recognition
+- MoviePy
+- FPDF
+
```
Comment (HarshCasper) [inline Python/Video to PDF(CLI)/README.md:17]: Add the command through which a user can install all the packages in a single go.

File: Python/Video to PDF(CLI)/video_to_pdf.py [hunk 19]
```diff
@@ -0,0 +1,78 @@
...
+audio_clip = ''
+
+def audio_to_pdf():
+    '''
+    Function to convert audio to pdf:
+    Here, we first extract the audio from the selected video and store it into
+    'my_audio.wav'. Then, using the speech recognition module and Google speech API,
+    we convert this audio to text and write it into a file named 'my_text.txt'.
+    This text file is then passed to text_to_pdf function to convert it into a pdf and at
+    the end the audio and text files are removed using the OS module as we don't need them anymore.
```
Comment (HarshCasper) [inline Python/Video to PDF(CLI)/video_to_pdf.py:19]: You don't explain how your code works in a docstring. Just mention the function params, return type and the purpose (which you already pitched)

File: Python/Video to PDF(CLI)/video_to_pdf.py [hunk 20]
```diff
@@ -0,0 +1,78 @@
...
+
+
+def text_to_pdf(file):
+    '''
+    Function to convert text to pdf:
+    In this function, we create a pdf using the FPDF module and add a page to it and set the font.
+    Effective page width is calculated so that
+    Then, we open the text file in read mode and using multi_cell(for wrapping the text into multiple lines),
+    insert the text from the text file into the pdf.
+    At last, we save it as 'my_pdf.pdf'.
```
Comment (HarshCasper) [inline Python/Video to PDF(CLI)/video_to_pdf.py:54]: You don't explain how your code works in a docstring. Just mention the function params, return type and the purpose (which you already pitched)

**Review by HarshCasper — CHANGES_REQUESTED** [review]

**Review by HarshCasper — APPROVED** [review]

**Pull request merged**